matsim/src/main/java/org/matsim/core/router/FallbackRoutingModule.java

```diff
@@ -0,0 +1,4 @@
+package org.matsim.core.router;
+
+public interface FallbackRoutingModule extends RoutingModule {
+}
```

Comment:
Do we really need this interface? It does not provide additional functionality. The fallback routing module is just a routing module that is used as a fallback. The fact that it is used as a fallback comes from its designation of a specific routing module, not from implementing a specific interface.

One could argue that the fallback routing module is not allowed to return `null`, compared to a regular routing module. But I don't see this constraint here either.

Reply:
Ah, now I see, dependency injection... It's easier to bind a specific implementation to this specific interface, so it will be used as fallback routing module. I understand it now, but it kind of confirms my opinion that dependency injection does not necessarily help to write clean code, because clearly this interface is only useful due to dependency injection being used, and is kind of useless otherwise.

Reply:
The main reason for this actually was not that I wanted this to be replaceable, but that even the simple `FallbackRouterDefaultImpl` needs access to material (`Population`, `Network`) that `TripRouter` does not have.  And I didn't really want to touch the arguments of the public constructor of `TripRouter`.  So in that sense I would say that DI is in fact not so bad here since it allows each `RoutingModule` (including this one here) to use arbitrary information without having to pass it through `TripRouter` explicitly.